app/assets/javascripts/events.js.coffee

```diff
@@ -1,13 +1,60 @@
+
 jQuery ->
   $("#event_title").focus();
-  $('#events').dataTable
+  $table = $('#events').dataTable
+    "dom": '<"eventToolbar dataTables_filter">frtip',
     order: [[3, 'asc']],
     columnDefs: [
                   { orderable: false, targets: -1 },
                   { orderData: 2, targets: 3 },
                   { orderData: 4, targets: 5 },
-                  { visible: false, targets: [2,4] }
+                  { visible: false, targets: [2,4,8] }
                 ]
+# sets checkbox elements within #events dataTable()
+  $('div.eventToolbar').html('<label>Current:<input type="checkbox" id="js-events-current-checkbox" name="event-filter"></label>
+  <label>Past:<input type="checkbox" id="js-events-past-checkbox" name="event-filter"></label>
+  <label>Templates:<input type="checkbox" id="js-events-template-checkbox" name="event-filter"></label>')
+
+# logic handling checkbox filtering with dataTables()
+  filterCheckboxes = (element, criteria) ->
+    $.fn.dataTable.ext.search.push((settings, data) ->
+      checkboxState = element.checked
+      endTime = new Date(data[5])
+      isTemplate = data[8]
+
+      if criteria.type is "current" and (checkboxState is false or (checkboxState is true and endTime > criteria.dateTime))
```

Comment:
I would switch this to a switch statement on criteria.type, returning the boolean check on the checkboxState and defaulting to False

Reply:
@MadonnaMat with Kevin's requests I ended up completely refactoring this function so I will be sending the new pull request soon for a new assessment. thanks.